power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -503,9 +503,11 @@ template <bool sym> class YBus {
     uint64_t register_parameters_changed_callback(ParamChangedCallback callback) {
```

Comment:
I also got some complains (clang-tidy) about the callback not being const &.

Reply:
oh that sucks... good that you mention it... let me think about it for a second because i'm not sure what happens if i make it `const&` due to usage of `reference_wrapper`

Reply:
how do you get the `clang-tidy` warnings? i'm not getting any locally

Reply:
I think it's due to the order that I processed the smells: I first change the `insert` to `emplace_back` as SonarCloud suggests, then the `clang-tidy` complaint about `const &` starts. With the newest pull I don't get them anymore.